src/http_crawler/__init__.py

```diff
@@ -19,7 +19,10 @@ def crawl(base_url, follow_external_links=True):
 
     while todo:
         url = todo.pop()
-        rsp = session.get(url)
+        rsp = session.get(url, allow_redirects=follow_redirects)
+
+        if rsp.status_code // 100 == 3:
+            continue
```

Comment:
> add a check in this section so that if the response is a 3xx, we try to extract a URL from the Location header.

By doing this, wouldn't we essentially be _following_ redirects? I am confused. 

If we are proceeding this manner, I think it would be a better approach to scrap the body of the 3xx HTTP response also, in addition to adding the url from `location` header.